test/tutorial_vector.jl

```diff
@@ -0,0 +1,78 @@
+# Tests based on the Vector API tutorial found at http://gdal.org/ogr_apitut.html
+
+
+# Reading From OGR
+
+nil = convert(Ptr{GDAL.Cstring}, C_NULL)
+
+# TODO: provide a convenient way to construct nOpenFlags
+# nOpenFlags: a combination of GDAL_OF_ flags that may be combined through logical or operator
+# Driver kind: GDAL_OF_RASTER for raster drivers, GDAL_OF_VECTOR for vector drivers. If none of the value is specified, both kinds are implied.
+# Access mode: GDAL_OF_READONLY (exclusive)or GDAL_OF_UPDATE.
+# Shared mode: GDAL_OF_SHARED. If set, it allows the sharing of GDALDataset handles for a dataset with other callers that have set GDAL_OF_SHARED. In particular, GDALOpenEx() will first consult its list of currently open and shared GDALDataset's, and if the GetDescription() name for one exactly matches the pszFilename passed to GDALOpenEx() it will be referenced and returned, if GDALOpenEx() is called from the same thread.
+# Verbose error: GDAL_OF_VERBOSE_ERROR. If set, a failed attempt to open the file will lead to an error message to be reported.
+
+# file from GDAL SVN
+dataset = GDAL.openex("data/point.geojson", UInt32(GDAL.GDAL_OF_VECTOR), nil, nil, nil)
```

Comment:
What are you proposing? Compared to master it is already relaxed in this branch:
https://github.com/visr/GDAL.jl/blob/d76caa2263fbd64560bb1057a11698e3b925b9ee/src/C/gdal.jl#L313-L315